aspnetcore/blazor/components/index.md

```diff
@@ -1302,9 +1302,7 @@ A component reference is only populated after the component is rendered and its
 
 To manipulate component references after the component has finished rendering, use the [`OnAfterRender` or `OnAfterRenderAsync` methods](xref:blazor/components/lifecycle#after-component-render-onafterrenderasync).
 
-To use a reference variable with an event handler, use a lambda expression or assign the event handler delegate in the [`OnAfterRender` or `OnAfterRenderAsync` methods](xref:blazor/components/lifecycle#after-component-render-onafterrenderasync). This ensures that the reference variable is assigned before the event handler is assigned.
-
-The following lambda approach uses the preceding `ReferenceChild` component.
+The following lambda approach uses the preceding `ReferenceChild` component. The [`!` (null-forgiving) operator](/dotnet/csharp/language-reference/operators/null-forgiving) results in a clear error if `childComponent` isn't populated when `ChildMethod` is called, but `childComponent` is populated in all normal Blazor framework rendering scenarios. Don't attempt to call a referenced component method without a lambda expression (for example, `@onclick="childComponent!.ChildMethod(5)"`) because the reference variable may not be assigned at the time click event is assigned.
```

Comment:
The null-forgiving operator (`!') has no effect at run time.
Let me suggest a change, will be back in a minute.

Reply:
Well ... it shouldn't have any affect anyway according to Steve.

If we change this, then I'll need to ping him because we'll be moving away from his sentiment AFAICT. I'll wait to see your suggestion.

Reply:
Sorry, the tooling didn't allow me to suggest code changes to this PR, so I submitted a new PR: #31868 😇

Reply:
Use quote markdown in that case ... but I don't understand why GH is being such a pain this morning 😄.